js/core.js

```diff
@@ -177,6 +177,8 @@ var ERMrest = (function (module) {
                 var catalog = new Catalog(self._server, id);
                 catalog._introspect().then(function () {
                     self._catalogs[id] = catalog;
```

Comment:
Recall that I said you have to defer this assignment until after you are sure everything has been resolved without errors. So you need to move this line down to in between line 181 & 182, just before you resolve the promise.